sympy/sets/tests/test_conditionset.py

```diff
@@ -30,52 +31,78 @@ def test_CondSet():
     # not depend on the dummy symbol): the result is `y > 5`.
     # In this case, ConditionSet is just acting like
     # Piecewise((Interval(1, 7), y > 5), (S.EmptySet, True)).
-    raises(TypeError, lambda: 6 in ConditionSet(x, y > 5, Interval(1, 7)))
+    raises(TypeError, lambda: 6 in ConditionSet(x, y > 5,
+        Interval(1, 7)))
 
-    assert isinstance(ConditionSet(x, x < 1, {x, y}).base_set, FiniteSet)
+    assert isinstance(ConditionSet(x, x < 1, {x, y}).base_set,
+        FiniteSet)
     raises(TypeError, lambda: ConditionSet(x, x + 1, {x, y}))
     raises(TypeError, lambda: ConditionSet(x, x, 1))
+    raises(BadSignatureError, lambda: ConditionSet((x, x), x))
 
     I = S.Integers
     C = ConditionSet
+    assert C(x, False, I) is S.EmptySet
+    assert C(x, True, I) is I
     assert C(x, x < 1, C(x, x < 2, I)
         ) == C(x, (x < 1) & (x < 2), I)
     assert C(y, y < 1, C(x, y < 2, I)
-        ) == C(x, (x < 1) & (y < 2), I)
+        ) == C(x, (x < 1) & (y < 2), I), C(y, y < 1, C(x, y < 2, I))
     assert C(y, y < 1, C(x, x < 2, I)
         ) == C(y, (y < 1) & (y < 2), I)
     assert C(y, y < 1, C(x, y < x, I)
         ) == C(x, (x < 1) & (y < x), I)
-    assert C(y, x < 1, C(x, y < x, I)
-        ) == C(L, (x < 1) & (y < L), I)
-    c = C(y, x < 1, C(x, L < y, I))
-    assert c == C(c.sym, (L < y) & (x < 1), I)
-    assert c.sym not in (x, y, L)
-    c = C(y, x < 1, C(x, y < x, FiniteSet(L)))
-    assert c == C(L, And(x < 1, y < L), FiniteSet(L))
+    assert unchanged(C, y, x < 1, C(x, y < x, I))
```

Comment:
This example is really confusing. On master we have
```
In [23]: C(y, x < 1, C(x, y < x, Integers))                                                                                                                   
Out[23]: {λ | λ ∊ ℤ ∧ (x < 1 ∧ y < λ)}
```
Is that wrong?

I suppose y in the outer is bound but y in the inner is free. The bound x in the inner shouldn't be matched with the free x in the outer so `y < λ` isn't right...

Reply:
I simply removed the attempt to resolve the confusing possibility of clash between symbols in the two ConditionSets. Now no symbols are changed. It was going to get even more confusing to untangle this for Tuple-args.

Reply:
`s = C(x, y < x, Integers) -> C(z, z > y, Integers)`
`C(y, x < 1, s) -> C(z, x < 1, s)`
so `C(z, x < 1, C(z, z > y, Integers))` -> `C(z, And(x < 1, z > y), Integers)` 

Although this no longer rewrites like this, is there a value that you can think of that would make it fail?

Reply:
> is there a value that you can think of that would make it fail?

No, I think your inference is correct.

I just realised that the original test example is even more confusing because the base set is `{L}` and the symbol changes to `L` giving
```
ConditionSet(L, ..., FiniteSet(L))
```
so L is both free and bound in that result...